src/Core/src/Platform/iOS/FlyoutViewExtensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿using CoreGraphics;
+using UIKit;
+
+namespace Microsoft.Maui.Platform;
+
+internal static class FlyoutViewExtensions
+{
+	internal static UIImage? AutoResizeFlyoutIcon(UIImage image)
+	{
+		if (image == null)
+		{
+			return null;
+		}
+
+		CGSize newSize = image.Size;
+
+		if (image.Size.Width > image.Size.Height) //Wide
+		{
+			newSize.Width = 24;
+			newSize.Height = newSize.Width * image.Size.Height / image.Size.Width;
+		}
+		else if (image.Size.Width < image.Size.Height) //Tall
+		{
+			newSize.Height = 24;
+			newSize.Width = newSize.Height * image.Size.Width / image.Size.Height;
+		}
+		else //Square
+		{
+			newSize.Width = 24;
+			newSize.Height = newSize.Width;
+		}
+
+		UIGraphics.BeginImageContextWithOptions(newSize, false, 0);
```

Comment:
Wrap in a try-finally block to ensure that `UIGraphics.EndImageContext()` is always executed.